src/main/java/calculator/StringCalculator.java

```diff
@@ -0,0 +1,82 @@
+package calculator;
+
+import java.util.StringTokenizer;
+
+import static java.lang.Double.parseDouble;
+
+public class StringCalculator {
+
+    public double calculate(String text) {
+        if (text.equals(" ") || text.isEmpty()) {
+            throw new EmptyExpressionException();
+        }
+
+        StringTokenizer stringTokenizer = new StringTokenizer(text);
+
+        if (isPossibleToCalulateFirst(stringTokenizer)) {
+            throw new WrongExpressionException();
+        }
+
+        double result = calculate(stringTokenizer);
+
+        isNotInfinity(result);
+
+        if (!stringTokenizer.hasMoreTokens()) {
+            return result;
+        }
+
+        return nextCalculate(result, stringTokenizer);
+    }
+
+    private double calculate(StringTokenizer stringTokenizer) {
+        return OperatorWith.of(getDoubleValue(stringTokenizer.nextToken()),
+                getOperator(stringTokenizer.nextToken()),
+                getDoubleValue(stringTokenizer.nextToken()))
+                .calculate();
```

Comment:
https://github.com/next-step/java-racingcar/pull/2273#discussion_r672903805 에 대한 제 생각을 드리자면 

```suggestion
        return getOperator(stringTokenizer.nextToken())
                .calculate(getDoubleValue(stringTokenizer.nextToken()), stringTokenizer.nextToken()));
```

의 코드와 동일한 기능을 하고 있다는 생각이 드는데요!
즉, `OperatorWith` 클래스에서 어떠한 역할을 담당하고 있는지 파악하기가 힘들어 남긴 리뷰였습니다~
객체가 존재하기 위해선 어떠한 역할과 책임이 존재하게 되는데, 현재 `OperatorWith` 클래스의 경우 단순 by-pass만 담당하고 있는 것 같아요!

`OperatorWith`에 조금 더 명확한 역할과 책임을 부여해보면 어떨까요!?
어려운 부분이 있거나 다른 의견이 있으면 편하게 말씀해주세요 😄